### ISSUE TITLE
Strange indentation behavior

### ISSUE THREAD

**Issue by foxbunny** [posva/vim-vue#31]
With the following code:

```
<template>
  <div id="root">
    <app-display/>
    <app-increment/>
  </div>
</template>

<script>
  import AppDisplay from './display.vue';
  import AppIncrement from './increment.vue';
  import store from './store';

  export default {
    components: {AppDisplay, AppIncrement},
    store
  };
</script>
```

if I select it all and hit `=` to reindent, this is what I get:

```
<template>
  <div id="root">
    <app-display/>
      <app-increment/>
      </div>
    </template>

    <script>
  import AppDisplay from './display.vue';
import AppIncrement from './increment.vue';
import store from './store';

export default {
  components: {AppDisplay, AppIncrement},
    store
};
  </script>
```

The mess in the `<template>` tag seems to be due to the self-closing tags. If i close them it behaves correctly:

```
<template>
  <div id="root">
    <app-display></app-display>
    <app-increment></app-display>
  </div>
</template>
```

The `<script>` tag, on the other hand still does not indent correctly:

```
<script>
  import AppDisplay from './display.vue';
import AppIncrement from './increment.vue';
import store from './store';

export default {
  components: {AppDisplay, AppIncrement},
    store
};
  </script>
```

EDIT: forgot to mention this is on Vim 8 on Windows 10.

**Comment by foxbunny** [comment]
Another weird thing I noticed just now.

After closing the `<template>` tag, when I type `<script>` and hit Return, the `<script>` tag is automatically intended by one level. If I continue to type on the next line, and then reindent the opening `<script>` tag with `=`, then it goes back to the correct position.

**Comment by posva** [comment]
Tested with HTML5 plugin and it behaves the same way.
Since vim-vue relies on your installed html syntax plugin, it's on that end you should open the issue.
I recommend you opening an issue on [othree/html5.vim](https://github.com/othree/html5.vim)

**Comment by foxbunny** [comment]
After removing the HTML5 plugin, the indentation within the `<script>` tag remains unchanged:

```
<script>
  import AppDisplay from './display.vue';
import AppIncrement from './increment.vue';
import store from './store';

export default {
  components: {AppDisplay, AppIncrement},
    store
};
  </script>
```

**Comment by posva** [comment]
it's a problem with the html syntax. Try using `ft=html`, you'll see the same behaviour

**Comment by foxbunny** [comment]
Ah, so I get it now, vue just uses the same indentation rules as ft=html? If that is case, do you have a recommendation for how to set it up so it works acceptably well with vue?

**Comment by posva** [comment]
yep. I'll recommend fixing it in the html plugin so more people can benefit from that

**Comment by foxbunny** [comment]
Should I take that as not working for you either?

To be honest, the entire concept that this is being controlled by the html ft is wrong to begin with. But of course, either way it has nothing to do with vim-vue, so I'll just stfu. :)

**Comment by posva** [comment]
But if it doesn't rely on the html plugin it should parse the html syntaxt itself, shouldn't it?

**Comment by foxbunny** [comment]
What I meant is, if it's JavaScript, vim should provide a mechanism to say something like "everything within `<script>` should be treated as `ft=javascript` or something along those lines, and this problem goes away (probably).

**Comment by posva** [comment]
It works a bit like that but the mechanism isn't perfect 😓

**Comment by foxbunny** [comment]
I can see that. :D

**Comment by scottgrayson** [comment]
is there a workaround for this?

**Comment by posva** [comment]
indent progressively by lines

**Comment by scottgrayson** [comment]
https://github.com/scottgrayson/Dotfiles/blob/master/vimrc#L111

not the best at vimscript, but fixes this code block
```
<script>
  import AppDisplay from './display.vue';
import AppIncrement from './increment.vue';
import store from './store';

export default {
  components: {AppDisplay, AppIncrement},
    store
};
  </script>
```

**Comment by desprit** [comment]
I also have some strange indent behavior in my vue file:

```
<script>
  export default {
    data () {
      return: {
        someVar: {*cursor_is_here*},
      }
    }
  }
</script>
```

Now when I hit enter I expect it to be:
```
<script>
  export default {
    data () {
      return: {
        someVar: {
          *cursor_is_here*
        },
      }
    }
  }
</script>
```

but instead I get this:
```
<script>
  export default {
    data () {
      return: {
someVar: {
*cursor_is_here*
        },
      }
    }
  }
</script>
```

**Comment by adriaanzon** [comment]
This got fixed in d4793f59631f601a698f5f3117c58a9b007cca20 (except for self-closing tags), can you confirm it works?

**Comment by tomc1998** [comment]
Is there any update on the self closing tags issue, or any follow up issues in the HTML plugin? It's super frustrating, I'm basically just indenting manually at the moment! Or rather, is there any way to make vim-vue use the `xml` filetype for the <template> tags? That fixes my issue, and seems to work just about fine for me.
